test/Database/LSMTree/ModelIO/Normal.hs

```diff
@@ -201,8 +202,19 @@ snapshot ::
   -> TableHandle m k v blob
   -> m ()
 snapshot n TableHandle {..} = atomically $
-    withModel "snapshot" thSession thRef $ \tbl ->
-        modifyTVar' (snapshots thSession) (Map.insert n (toDyn tbl))
+    withModel "snapshot" thSession thRef $ \tbl -> do
```

Comment:
This is not defined behaviour. The spec for `snapshot` doesn't say whether it should or shouldn't error if the snapshot with that name already exists.

@dcoutts should clarify.

Reply:
For example,

```
-- * It is safe to concurrently make snapshots from any table, provided that
--   the snapshot names are distinct (otherwise this would be a race).
```

which doesn't imply either way. The race may end with "latest wins" or "latest gets an exception"

Reply:
Alright, we should in the specification what happens in this case. IMO, it is better to throw an exception, but @dcoutts  should chime in too

Reply:
I think both variants are useful. We might need to tweak API to allow both. There could also be a variant which returns `Maybe TableHandle`, i.e. kind of `swap`.

Reply:
I think we should encourage applications to manage their snapshots explicitly, and delete ones they don't need. We can't in any case make snapshot replacement be atomic. So I think it's better to have apps deal with deleting before replacing.

So yes, lets throw an exception if the snapshot already exists.